src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessible.java

```diff
@@ -172,7 +172,8 @@ public void propertyChange(PropertyChangeEvent e) {
                                 ((AccessibleState)oldValue) == AccessibleState.VISIBLE ) {
                             menuClosed(ptr);
                         }
-                    } else if (thisRole == AccessibleRole.MENU_ITEM) {
+                    } else if (thisRole == AccessibleRole.MENU_ITEM ||
+                            ((parentRole == AccessibleRole.POPUP_MENU) && (thisRole == AccessibleRole.MENU))) {
```

Comment:
Is the check for the AccessibleRole.POPUP_MENU is needed here? How it will work if the Menu is added to some container other than POPUP_MENU?

Reply:
done